bpf/sockops/bpf_sockops.c

```diff
@@ -48,20 +48,35 @@ static inline void bpf_sock_ops_ipv4(struct bpf_sock_ops *skops)
 {
 	__u32 dip4, dport, dstID = 0;
 	struct endpoint_info *exists;
-	struct lb4_key lb4_key = {};
 	struct sock_key key = {};
-	struct lb4_service *svc;
 	int verdict;
 
 	sk_extract4_key(skops, &key);
-	sk_lb4_key(&lb4_key, &key);
 
 	/* If endpoint a service use L4/L3 stack for now. These can be
 	 * pulled in as needed.
 	 */
-	svc = __lb4_lookup_service(&lb4_key);
-	if (svc)
-		return;
+#ifdef ENABLE_LEGACY_SERVICES
+	{
+		struct lb4_key lb4_key = {};
+		struct lb4_service *svc
+
+		sk_lb4_key(&lb4_key, &key);
+		svc = __lb4_lookup_service(&lb4_key);
+		if (svc)
+			return;
+	}
+#else
```

Comment:
This seems wrong, `ENABLE_LEGACY_SERVICES` shouldn't disable lookup of new maps.

Reply:
@brb ^^^

Reply:
OK in ./bpf/lib/lb.h ENABLE_LEGACY_SERVICES does not disable lookup on new maps so I'll convert that else into an end. So we check both v1 and v2 maps in legacy case.

Reply:
@brb the way I read bpf_lxc.c makes me think I can drop the legacy checks altogether?

        /* It's safe to lookup the v2 svc instead of the legacy, as the v2 is 
         * created before the endpoint datapath code has been reloaded. */
        if ((svc = lb4_lookup_service_v2(skb, &key)) != NULL) {